instance_test.go

```diff
@@ -0,0 +1,61 @@
+package thema_test
```

Comment:
Did the trick to avoid cyclic dependencies (_note it's using the `vmux` package, for instance, as well as some of `txtartest` packages that also rely on Thema's core_), but not sure how acceptable is it.

Reply:
In general i do support use of this pattern. In this particular case, though, my gut feel is that the base thema package should probably avoid reliance on any other packages for helpers in tests. That helps us ensure we're layering our tests effectively, and keeps test failure signals cleaner (i.e. `github.com/grafana/thema` tests cannot fail as a result of any refactoring or bugs in `gh/grafana/thema/vmux` package)

i wouldn't say this is a hard rule, but where the implementations in other packages are fairly trivial, i think it's preferable to copy over importing. In this case:

* `bindlin` already has an [unexported equivalent](https://github.com/grafana/thema/blob/53a6fb01cf3135287b2501a08b9381b5c9831bbd/lineage_test.go#L82) in thema pkg tests
* the JSON codec in `vmux` is a pretty trivial wrapper around calls into other CUE funcs. i'd say it's fine to make an unexported helper in tests, same as with the bindlin func

Reply:
Absolutely agree 💯 

> In general i do support use of this pattern. In this particular case, though, my gut feel is that the base thema package should probably avoid reliance on any other packages for helpers in tests. That helps us ensure we're layering our tests effectively, and keeps test failure signals cleaner (i.e. github.com/grafana/thema tests cannot fail as a result of any refactoring or bugs in gh/grafana/thema/vmux package)